solang-parser/src/diagnostics.rs

```diff
@@ -96,6 +97,29 @@ impl Diagnostic {
         }
     }
 
+    pub fn cast_error(loc: Loc, message: String) -> Self {
+        Diagnostic {
+            level: Level::Error,
+            ty: ErrorType::CastError,
+            loc,
+            message,
+            notes: Vec::new(),
+        }
+    }
+
+    pub fn cast_error_with_note(loc: Loc, message: String, note_loc: Loc, note: String) -> Self {
```

Comment:
I wonder if we should use `#[inline]` with these functions, because we are moving data three times: from the arguments to the `struct Diagnostic` and from the struct to the place that called the function.

Reply:
llvm will auto-inline functions based on some heuristics. I think `#[inline]` should really only be used if you want to force the compiler to inline, even though it makes the binary bigger, because of some measurement you've done which shows there is a benefit.